qmq-common/src/test/java/qunar/tc/qmq/utils/utils/RetrySubjectUtilsTest.java

```diff
@@ -0,0 +1,46 @@
+package qunar.tc.qmq.utils.utils;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import static qunar.tc.qmq.utils.RetrySubjectUtils.*;
+
+public class RetrySubjectUtilsTest {
+
+    @Test
+    public void testGetRealSubject() {
+        assertNull(getRealSubject(null));
+
+        assertEquals("%RETRY", getRealSubject("%RETRY"));
+        assertEquals("foo", getRealSubject("%DEAD_RETRY%foo%bar"));
+    }
+
+    @Test
+    public void testParseSubjectAndGroup() {
+        assertNull(parseSubjectAndGroup(null));
+        assertNull(parseSubjectAndGroup("%DEAD_RETRY"));
+
+        assertNotNull(parseSubjectAndGroup("%DEAD_RETRY%foo%bar"));
+    }
+
+    @Test
+    public void testIsRealSubject() {
+        assertFalse(isRealSubject(null));
+        assertFalse(isRealSubject("%RETRY"));
+        assertFalse(isRealSubject("%DEAD_RETRY"));
+
+        assertTrue(isRealSubject("foo"));
+        assertTrue(isRealSubject("bar"));
```

Comment:
How do these 3 test different parts of the logic?

Reply:
Removed tow of the three test as they were duplicates